tests/libs/common/common_tests.cpp

```diff
@@ -86,3 +90,70 @@ verify_utility_helper_results(_In_ const bpf_object* object)
         (test_data[1].timestamp - test_data[0].timestamp));
 }
+
+int
+ring_buffer_test_event_handler(void* ctx, _In_ void* data, size_t size)
+{
+    ring_buffer_test_event_context_t* event_context = reinterpret_cast<ring_buffer_test_event_context_t*>(ctx);
+    if ((event_context->cancelled) || (event_context->matched_entry_count == RING_BUFFER_TEST_EVENT_COUNT))
+        // Either ring buffer subscription is cancelled, or required number of event notifications already reserved.
+        // Simply return.
+        return 0;
+
+    std::vector<char> event_record(reinterpret_cast<char*>(data), reinterpret_cast<char*>(data) + size);
+    // Check if indicated event record matches an entry in the context app_ids list.
+    auto records = event_context->records;
+    auto it = std::find(records->begin(), records->end(), event_record);
+    if (it != records->end())
+        event_context->matched_entry_count++;
+    if (event_context->matched_entry_count == RING_BUFFER_TEST_EVENT_COUNT) {
+        // If all the entries in the app Id list was found, fulfill the promise.
+        auto promise = event_context->ring_buffer_event_promise;
+        promise->set_value();
+    }
+    return 0;
+}
+
+typedef struct _ebpf_ring_buffer_subscription ring_buffer_subscription_t;
+
+void
+ring_buffer_api_test_helper(
+    fd_t ring_buffer_map, std::vector<std::vector<char>> expected_records, std::function<void(int)> generate_event)
+{
+    // Ring buffer event callback context.
+    ring_buffer_test_event_context_t context{};
+    context.matched_entry_count = 0;
+
+    context.records = &expected_records;
+
+    // Generate events prior to subscribing for ring buffer events.
+    for (int i = 0; i < RING_BUFFER_TEST_EVENT_COUNT / 2; i++) {
+        generate_event(i);
+    }
+
+    // Associate a promise object with ring buffer event context, which should be completed
+    // once notifications for all events are received.
+    std::promise<void> ring_buffer_event_promise;
+    auto ring_buffer_event_callback = ring_buffer_event_promise.get_future();
+    context.ring_buffer_event_promise = &ring_buffer_event_promise;
+
+    // Create a new ring buffer manager and subscribe to ring buffer events.
+    // The notifications for the events that were generated before should occur after the subscribe call.
+    struct ring_buffer* ring_buffer =
+        ring_buffer__new(ring_buffer_map, ring_buffer_test_event_handler, &context, nullptr);
+    REQUIRE(ring_buffer != nullptr);
+
+    // Generate more events, post-subscription.
+    for (int i = RING_BUFFER_TEST_EVENT_COUNT / 2; i < RING_BUFFER_TEST_EVENT_COUNT; i++) {
+        generate_event(i);
+    }
+
+    // Wait for event handler getting notifications for all RING_BUFFER_TEST_EVENT_COUNT events.
+    REQUIRE(ring_buffer_event_callback.wait_for(1s) == std::future_status::ready);
+
+    // Mark the event context as cancelled, such that the loopback stops processing events.
```

Comment:
```suggestion
    // Mark the event context as canceled, such that the loopback stops processing events.
```
Also did you really mean "loopback" or should that be "loop"?

Reply:
loop. Fixed.